tests/functional/DocumentTest.php

```diff
@@ -966,4 +968,279 @@ public function testGetDuplicateOf()
+            'items_id' => $kb_id,
+        ]);
+        $this->assertCount(1, $datas);
+
+        $this->assertTrue($document->getFromDB(current($datas)['documents_id']));
+        $this->assertEquals(0, $document->fields['documentcategories_id']);
+
+
+        /////////////////////////////////////////////////////////////////////////////////////
+        // Update config to have default category for document uploaded during kb creation //
+        /////////////////////////////////////////////////////////////////////////////////////
+        $documentCategory_id = $documentCategory->add([
+            'name'        => 'Default Category',
+        ]);
+        $this->assertGreaterThan(0, $documentCategory_id);
+        $CFG_GLPI['documentcategories_id_forticket'] = $documentCategory_id;
+
+
+        /////////////////////////////////////////////////////////////////////////////////////////////
+        // Create KnowbaseItem with document, check document has category defined in configuration //
+        /////////////////////////////////////////////////////////////////////////////////////////////
+        $kb = new \KnowbaseItem();
+        $filename = 'azerty987654.444444' . 'foo2.txt';
+        $input2 = [
+            'name' => 'KnowbaseItem 2',
+            'content' => 'testUploadDocumentWithoutCategory',
+            '_filename' => [
+                $filename,
+            ],
+            '_tag_filename' => [
+                'abc123-def456-ghi789.2222',
+            ],
+            '_prefix_filename' => [
+                'azerty987654.444444',
+            ],
+        ];
+
+        copy(FIXTURE_DIR . '/uploads/bar.txt', GLPI_TMP_DIR . '/' . $filename);
+        $kb_id = $kb->add($input2);
+
+        // Load doc for ticket2
+        $datas = $document_item->find([
+            'itemtype' => \KnowbaseItem::class,
+            'items_id' => $kb_id,
+        ]);
+        $this->assertCount(1, $datas);
+
+        $this->assertTrue($document->getFromDB(current($datas)['documents_id']));
+        $this->assertEquals(0, $document->fields['documentcategories_id']);
+    }
+
+
+    public function testDefaultDocumentCategoryFromDocumentForm()
+    {
+        $this->login();
+
+        global $CFG_GLPI;
+
+        $document = new \Document();
+        $documentCategory = new DocumentCategory();
+
+        ///////////////////////////////////////////////////////////////////////////////////////
+        // Create Ticket, add new document via form with itemtype, check category is not set //
+        //////////////////////////////////////////////////////////////////////////////////////
+
+        $ticket = new \Ticket();
+        $input = [
+            'name' => 'Ticket 1',
+            'content' => 'testDefaultDocumentCategoryFromDocumentForm',
+            'entities_id' => 0,
+            'is_recursive' => 0,
+        ];
+
+        $ticket_id = $ticket->add($input);
+        $this->assertGreaterThan(0, $ticket_id);
```

Comment:
For creations in tests, prefer `createItem()` to `add()` because it checks more than `assertGreaterThan(...)`.
```suggestion
        $input = [
            'name' => 'Ticket 1',
            'content' => 'testDefaultDocumentCategoryFromDocumentForm',
            'entities_id' => 0,
            'is_recursive' => 0,
        ];

        $ticket_id = $this->createItem(Ticket::class, $input)->getID();
```
_This also applies to other cases._